### ISSUE TITLE
Superset 4.0.1 - Exported CSV Contains Garbled Chinese Characters and Numbers

### ISSUE THREAD

**Issue by glenn-gj-liang** [apache/superset#29224]
### Bug description

I have installed the latest version of Superset (4.0.1). However, when exporting query results to a CSV file, the Chinese characters and numbers in the file are always garbled. I have tried changing the encoding to utf-8, utf-8-sig, and gbk, but the issue persists.

### How to reproduce the bug

Install Superset 4.0.1.
Run a query that includes Chinese characters and numbers.
Export the query results to a CSV file.
Open the CSV file and observe that the Chinese characters and numbers are garbled.

### Superset version

4.0.1

### Python version

3.11

### Node version

20.12.2

### Browser

Chrome

### Additional context

Expected Behavior:
The Chinese characters and numbers should be displayed correctly in the exported CSV file.

Actual Behavior:
The Chinese characters and numbers are garbled in the exported CSV file.

Environment:

Superset version: 4.0.1
Encoding attempted: utf-8, utf-8-sig, gbk
Additional Context:
Please provide any additional information that might help in diagnosing the 

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
Related issue linked above ^^^

Might even be considered a duplicate, really, but they seem _kind of_ unique.

**Comment by TommyLeung-gj** [comment]
No ,It still doesn't work after I changed encoding = "utf-8"

**Comment by TommyLeung-gj** [comment]
maybe there is something wrong in the process of async query ,  the celery or redis resultbackend cache

**Comment by bionexit** [comment]
+1

**Comment by Habeeb556** [comment]
+2

even I added the below configuration to `superset_config.py`

```
# CSV Configuration

# CSV Options: key/value pairs that will be passed as argument to DataFrame.to_csv
# method.
# note: index option should not be overridden
CSV_EXPORT = {"encoding": "'utf-8-sig'"}
```

**Comment by Habeeb556** [comment]
+2

Having the same issue with `VERSION 4.01` even `VERSION 2.1.3` WORKING FINE WITH THESE PARAMETER CONFIG.

**Comment by Habeeb556** [comment]
The issue has been resolved by downgrading the package with the following command: `pip install Werkzeug==2.3.8`.

**Comment by bionexit** [comment]
> The issue has been resolved by downgrading the package with the following command: `pip install Werkzeug==2.3.8`.

I down grade the Werkzeug as 2.3.8 but no luck.
What's your encoding option?

My is following
```
CSV_EXPORT = {"encoding": "utf-8-sig"}
```

**Comment by Habeeb556** [comment]
> > The issue has been resolved by downgrading the package with the following command: `pip install Werkzeug==2.3.8`.
> 
> I down grade the Werkzeug as 2.3.8 but no luck. What's your encoding option?
> 
> My is following
> 
> ```
> CSV_EXPORT = {"encoding": "utf-8-sig"}
> ```

Yes, this encoding. But did you face the same problem with `VERSION 2.1.3` or `3.1.3`? Also I got a Chinese characters not with the English.

**Comment by bionexit** [comment]
> > > The issue has been resolved by downgrading the package with the following command: `pip install Werkzeug==2.3.8`.
> > 
> > 
> > I down grade the Werkzeug as 2.3.8 but no luck. What's your encoding option?
> > My is following
> > ```
> > CSV_EXPORT = {"encoding": "utf-8-sig"}
> > ```
> 
> Yes, this encoding. But did you face the same problem with `VERSION 2.1.3` or `3.1.3`? Also I got a Chinese characters not with the English.

It's worked after i reload the cerely service. Thanks a lot bro.

**Comment by Habeeb556** [comment]
++ @TommyLeung-gj, could you confirm if this downgrade solves your issue or not? Also, what language are you using?

++ @bionexit, we appreciate your feedback on the language characters you encountered, to report to the [Werkzeug team](https://github.com/pallets/werkzeug/issues/2922).

**Comment by foretony5211** [comment]
> > > > The issue has been resolved by downgrading the package with the following command: `pip install Werkzeug==2.3.8`.
> > > 
> > > 
> > > I down grade the Werkzeug as 2.3.8 but no luck. What's your encoding option?
> > > My is following
> > > ```
> > > CSV_EXPORT = {"encoding": "utf-8-sig"}
> > > ```
> > 
> > 
> > Yes, this encoding. But did you face the same problem with `VERSION 2.1.3` or `3.1.3`? Also I got a Chinese characters not with the English.
> 
> It's worked after i reload the cerely service. Thanks a lot bro.

I use docker. How to reload the cerely service. thanks.

**Comment by wuqicyber** [comment]
> ++ @TommyLeung-gj, could you confirm if this downgrade solves your issue or not? Also, what language are you using?
> 
> ++ [@bionexit](https://github.com/bionexit), we appreciate your feedback on the language characters you encountered, to report to the [Werkzeug team](https://github.com/pallets/werkzeug/issues/2922).

i've try downgraded Werkzeug==2.3.8, and it works, thanks

**Comment by rusackas** [comment]
Related, I think: https://github.com/apache/superset/pull/33720

Anyone able to reproduce thison 5.0.0 release candidates or on `master` branch?

**Comment by Habeeb556** [comment]
> Related, I think: [#33720](https://github.com/apache/superset/pull/33720)
> 
> Anyone able to reproduce thison 5.0.0 release candidates or on `master` branch?
 
This PR (#33720) resolves or is related to #19388. However, it is **not** related to #29410 or #29224, which are specifically about CSV issues in the codebase, not in SQL Lab.

I can still reproduce this issue with version `5.0.0` on a fresh installation. The problem is only resolved when I downgrade to `Werkzeug==2.3.8`.

**Comment by rusackas** [comment]
OK, thanks! Reopening for now, then.
